sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+      {
+        desired_found = true;
+        AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+        break;
+      }
+      else
+      {
+        // else ignore token.
+        AZ_RETURN_IF_FAILED(az_json_parser_skip_children(&jp));
+      }
+
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+    }
+  }
+  else
+  {
+    desired_found = true;
+  }
+
+  if (!desired_found)
+  {
+    printf("Desired property object not found in twin");
+    return AZ_ERROR_ITEM_NOT_FOUND;
+  }
+
+  if (jp.token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+  }
+  AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+
+  bool temp_found = false;
+  bool version_found = false;
+  while (!(temp_found && version_found) || (jp.token.kind != AZ_JSON_TOKEN_END_OBJECT))
+  {
+    if (az_json_token_is_text_equal(&jp.token, desired_temp_property_name))
+    {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&jp.token, parsed_value));
+      temp_found = true;
+    }
+    else if (az_json_token_is_text_equal(&jp.token, AZ_SPAN_FROM_STR("$version")))
+    {
+      AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+
+      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&jp.token, (uint32_t*)version_number));
+      version_found = true;
+    }
+    else
+    {
+      // else ignore token.
+      AZ_RETURN_IF_FAILED(az_json_parser_skip_children(&jp));
+    }
+    AZ_RETURN_IF_FAILED(az_json_parser_next_token(&jp));
+  }
+  
+  if(temp_found && version_found)
+  {
+    return AZ_OK;
+  }
+
+  return AZ_ERROR_ITEM_NOT_FOUND;
+}
+
+static bool update_device_temp(int32_t temp)
+{
+  current_device_temp = temp;
+
+  bool ret = false;
+  if (current_device_temp > device_max_temp)
+  {
+    device_max_temp = current_device_temp;
+    ret = true;
+  }
+  if (current_device_temp < device_min_temp)
+  {
+    device_min_temp = current_device_temp;
+  }
+
+  // Increment the avg count, add the new temp to the total, and calculate the new avg
+  device_temperature_avg_count++;
+  device_temperature_avg_total += current_device_temp;
+  device_avg_temp = device_temperature_avg_total / (int32_t)device_temperature_avg_count;
+
+  return ret;
+}
+
+// Switch on the type of twin message and handle accordingly | On desired prop, respond with max
+// temp reported prop.
+static void handle_twin_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_twin_response* twin_response)
+{
+  az_result result;
+  uint32_t desired_temp;
+  int32_t  version_num;
+  az_span twin_payload_span
+      = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+  // Determine what type of incoming twin message this is. Print relevant data for the message.
+  switch (twin_response->response_type)
+  {
+    // A response from a twin GET publish message with the twin document as a payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
```

Comment:
>AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET [](start = 9, length = 36)

Can (or I guess should) the code in the _GET label and in the _DESIRED_PROPERTIES be moved into a common function, with the is_twin_get passed in there?  

I get you don't have a lot of code here so dup isn't that bad, but if an OEM takes this as their starting tempalte I'm worried when things get more complex they're on path of duplication where we could show them that these are almost identical code paths.

I'd recommend consolidation to a helper.